skiko/src/awtMain/kotlin/org/jetbrains/skiko/swing/SkiaSwingLayer.kt

```diff
@@ -0,0 +1,123 @@
+package org.jetbrains.skiko.swing
+
+import org.jetbrains.skia.Canvas
+import org.jetbrains.skiko.*
+import org.jetbrains.skiko.redrawer.RedrawerManager
+import java.awt.Graphics2D
+import java.awt.GraphicsConfiguration
+import javax.accessibility.Accessible
+import javax.swing.SwingUtilities.isEventDispatchThread
+
+/**
+ * Swing component that draws content provided by [skikoView] with GPU acceleration using Skia engine.
+ *
+ * Drawn content can be clipped by providing [ClipRectangle] to [clipComponents].
+ *
+ * This component can be used for better interop with Swing,
+ * so all Swing functionality like z-ordering, double-buffering etc. will be taken into account during rendering.
+ *
+ * But if no interop with Swing is needed, it is better to use [SkiaLayer] instead.
+ */
+@Suppress("unused") // used in Compose Multiplatform
+@ExperimentalSkikoApi
+open class SkiaSwingLayer(
+    skikoView: SkikoView,
+    analytics: SkiaLayerAnalytics = SkiaLayerAnalytics.Empty,
+) : SkiaSwingLayerComponent() {
```

Comment:
Do we need `SkiaSwingLayerComponent`? Can we extend `JComponent` directly?

Reply:
As I told here https://github.com/JetBrains/skiko/pull/720#issuecomment-1594757816
We don't need it's just a contract of what Compose uses from SkiaSwingLayer to not to have unused declarations.
But I think to reduce complexity I will remove it and suppress errors

Reply:
Ah, missed this comment. I would remove it.

> I need to suppress "not-used" inspection on public methods

This is for all public class/functions. There should be another way to suppress this inspection (but @Suppress isn't the solution as well).